File: verification/council/nmc_adapter.go
package council

import (
	"context"
	"strings"
)

type NMCRegistryAdapter struct{}

func NewNMCRegistryAdapter() *NMCRegistryAdapter {
	return &NMCRegistryAdapter{}
}

func (a *NMCRegistryAdapter) Verify(ctx context.Context, registrationNumber string, council string) (*CouncilVerificationResult, error) {
	cleanReg := strings.TrimSpace(registrationNumber)
	
	// Simulated National Medical Commission (NMC) India / State Council Registry Adapter
	if cleanReg == "" {
		return &CouncilVerificationResult{
			IsVerified:         false,
			Status:             "NOT_FOUND",
			VerificationSource: "NMC_NATIONAL_REGISTER",
			Remarks:            "Empty registration number provided",
		}, nil
	}

	// Mock simulation check for invalid/suspended registration test cases
	if strings.HasSuffix(cleanReg, "999") {
		return &CouncilVerificationResult{
			IsVerified:         false,
			RegistrationNumber: cleanReg,
			CouncilName:        council,
			Status:             "SUSPENDED",
			VerificationSource: "NMC_NATIONAL_REGISTER",
			Remarks:            "Registration is suspended or blacklisted by Medical Council",
		}, nil
	}

	return &CouncilVerificationResult{
		IsVerified:         true,
		DoctorName:         "Dr. Rahul Kumar",
		RegistrationNumber: cleanReg,
		CouncilName:        council,
		RegistrationYear:   2021,
		Status:             "ACTIVE",
		VerificationSource: "NMC_NATIONAL_REGISTER",
		Remarks:            "Verified in National Medical Register",
	}, nil
}

func (a *NMCRegistryAdapter) GetProviderName() string {
	return "NMC_NATIONAL_REGISTER_ADAPTER"
}

File: python_backend/verification/council/nmc_adapter.py
from .provider import CouncilVerificationProvider, CouncilVerificationResult

class NMCRegistryAdapter(CouncilVerificationProvider):
    def verify(self, ctx: dict, registration_number: str, council: str) -> CouncilVerificationResult:
        clean_reg = registration_number.strip()
        
        if not clean_reg:
            return CouncilVerificationResult(
                is_verified=False,
                status="NOT_FOUND",
                verification_source="NMC_NATIONAL_REGISTER",
                remarks="Empty registration number provided"
            )

        if clean_reg.endswith("999"):
            return CouncilVerificationResult(
                is_verified=False,
                registration_number=clean_reg,
                council_name=council,
                status="SUSPENDED",
                verification_source="NMC_NATIONAL_REGISTER",
                remarks="Registration is suspended or blacklisted by Medical Council"
            )

        return CouncilVerificationResult(
            is_verified=True,
            doctor_name="Dr. Rahul Kumar",
            registration_number=clean_reg,
            council_name=council,
            registration_year=2021,
            status="ACTIVE",
            verification_source="NMC_NATIONAL_REGISTER",
            remarks="Verified in National Medical Register"
        )

    def get_provider_name(self) -> str:
        return "NMC_NATIONAL_REGISTER_ADAPTER"
